conda_forge_tick/auto_tick.py

```diff
@@ -477,6 +482,64 @@ def migration_factory(
         )
 
 
+def _outside_pin_range(pin_spec, current_pin, new_version):
+    pin_level = len(pin_spec.split('.'))
+    current_split = current_pin.split('.')
+    new_split = new_version.split('.')
+    # if our pin spec is more exact than our current pin then rebuild more precisely
+    if pin_level > len(current_split):
+        return True
+    for i in range(pin_level):
+        if current_split[i] != new_split[i]:
+            return True
+    return False
+
+
+def create_migration_yaml_creator(migrators: MutableSequence[Migrator], gx: nx.DiGraph):
+    with indir("../conda-forge-pinning-feedstock/recipe"):
+        pinnings = parse_config_file('conda_build_config.yaml',
+                                     config=Config(**CB_CONFIG))
+    for k, package_pin_list in pinnings.items():
+        # exclude non-package keys
+        if k not in gx.nodes:
+            # conda_build_config.yaml can't have `-` unlike our package names
+            k = k.replace('_', '-')
+
+        if (k in gx.nodes) and not gx.nodes[k]['payload'].get('archived', False) and gx.nodes[k]['payload'].get('version'):
```

Comment:
This will still fail for the multiple outputs case. We need to map k back through the outputs to feedstock mapping and check for that. If yes, we migrate the whole feedstock.